src/vsg/viewer/Viewer.cpp

```diff
@@ -282,21 +291,12 @@ void Viewer::compile(BufferPreferences bufferPreferences)
     {
         std::set<Device*> devices;
 
-        for (auto& commandGraph : task->commandGraphs)
+        for (auto& commandGraph : reverse(task->commandGraphs))
```

Comment:
Why is the reverse required here?

Reply:
to create pipeline in primary before used in secondaries

Reply:
I would have push primaries front but the current chosen design in submitTask mandates secondaries to be treated first (to track their primaries)

Reply:
i just fixed this poor design